src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -541,17 +541,29 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
 
   def typedApply(tree: untpd.Apply, pt: Type)(implicit ctx: Context): Tree = {
 
+    /** Try same application with an implicit inserted around the qualifier of the function
+     *  part. Return an optional value to indicate success.
+     */
+    def tryWithImplicitOnQualifier(fun1: Tree, proto: FunProto)(implicit ctx: Context): Option[Tree] =
+      tryInsertImplicitOnQualifier(fun1, proto) flatMap { fun2 =>
+        tryEither { implicit ctx =>
+          Some(typedApply(
+            cpy.Apply(tree)(untpd.TypedSplice(fun2), proto.typedArgs map untpd.TypedSplice),
+            pt)): Option[Tree]
+        } { (_, _) => None }
+     }
+
     def realApply(implicit ctx: Context): Tree = track("realApply") {
-      var proto = new FunProto(tree.args, IgnoredProto(pt), this)(argCtx(tree))
-      val fun1 = typedExpr(tree.fun, proto)
+      val originalProto = new FunProto(tree.args, IgnoredProto(pt), this)(argCtx(tree))
+      val fun1 = typedExpr(tree.fun, originalProto)
 
-      // Warning: The following line is dirty and fragile. We record that auto-tupling was demanded as
+      // Warning: The following lines are dirty and fragile. We record that auto-tupling was demanded as
       // a side effect in adapt. If it was, we assume the tupled proto-type in the rest of the application.
```

Comment:
Shouldn't this comment be updated? It says "we assume the tupled proto-type in the rest of the application" but we use both `proto` and `originalProto` below